lib/ukcpio/cpio.c

```diff
@@ -70,8 +108,17 @@ extract_file(const char *path, const char *contents, size_t len,
 
 	uk_pr_info("Extracting %s (%zu bytes)\n", path, len);
 
-	fd = uk_syscall_r_open(path, O_CREAT | O_WRONLY | O_TRUNC, 0);
-	if (fd < 0) {
+	fd = uk_syscall_r_open(path, O_CREAT | O_WRONLY | O_EXCL, 0);
+	if (unlikely(fd == -EEXIST)) {
+		uk_pr_info("Path exists, trying removal\n");
```

Comment:
just "removal" feels dishonest :thinking: maybe "overwrite"?

Reply:
I feel "overwrite" is misleading w.r.t. the operation being attempted: actual unlink/rmdir/wtvr, followed by extraction (the latter of which is explicitly said by the very previous info msg). This is relevant because "overwriting" a path is ambiguous: if the path is a symlink reasonable minds will disagree as to what "overwriting" that symlink means in casual conversation: whether to replace the symlink or what it points to. Indeed `open` without `O_EXCL` does the latter, and it's explicitly not what we want to happen.
Thus I want to be explicit that we're "removing" path before the extraction happens. That has pretty clear semantics.